caproto/_circuit.py

```diff
@@ -322,6 +324,16 @@ def _process_command(self, role, command):
                 raise err("priority {} does not match previously set priority "
                           "of {} for this circuit".format(command.priority,
                                                           self.priority))
+            protocol_version = min(self.protocol_version, command.version)
+            self.protocol_version = protocol_version
+            for cid, chan in self.channels.items():
```

Comment:
I think we should not bother with this loop (and the one like it below). If the same circuit gives us different VersionResponses, we should either leave existing channels alone or blow up.

Reply:
I'm arguing that it doesn't make sense for the same circuit to have more than one version. It's an open question whether any existing servers (that we need to interact with) violate this reasonable assumption anyway....